### ISSUE TITLE
Clean up headers with IWYU

### ISSUE THREAD

**Issue by ibell** [CoolProp/CoolProp#983]
@jowr - here is a fun project : cleaning up the headers so we remove superfluous includes.  See:

https://github.com/jasonmp85/homebrew-iwyu (for OSX)
http://stackoverflow.com/a/30951493/1360263 (cmake integration)

**Comment by ibell** [comment]
I got a lot of output when I did:

```
cmake ../.. -DCOOLPROP_MAIN_MODULE=ON -DCOOLPROP_IWYU=ON
cmake --build .
```

**Comment by jowr** [comment]
Good idea, it is great to have these little helpers that make you tidy up your code :)

**Comment by ibell** [comment]
I'm not sure we want to follow all their recommendations (I'm sure we don't), but it could help us tease out where we are including big headers (Eigen, etc.) when it is not necessary.

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
